### ISSUE TITLE
Webhook stream should continue feeding data - duplicates should be handled in Dashboard

### ISSUE THREAD

**Issue by cmanda** [rathnapandi/runners#18]
https://github.com/rathnapandi/runners/blob/51bd6032e314151dc266d7a9be23fe9b378af16c/src/main/java/com/axway/runners/strava/CallbackController.java#L153

**Comment by rathnapandi** [comment]
Can you give an example? The duplicate check is done on the field activityId as activityId is unique. If given activityId is stored in Elastic search it wont get stored again.

**Comment by cmanda** [comment]
The feed from Strava comes with obj_type indicating the **athlete/activity**, and **aspect_type=create/update/delete**

For e.g. 

- if athlete & Create are the event metadata, then it indicates a new athlete has signed up (OAuth consent trigger), and likewise 
- athlete & delete indicates the revoke access event happened. 
- Similarly, Activity & Create should POST to Feed, and Activity + Update should PUT to Feed, and Activity+ Delete should DELETE from Feed. This way the data model is always consistent with actual strava event.

Back to the enhancement, under the **Event & NonEvent Activity Feed flow, you should consider the Activity + Aspect Type - ideal case**, or, _Activity + Aspect Type (POST for CREATE/UPDATE) and let Kibana count only the most recent POSTed event (workaround)_

**Comment by rathnapandi** [comment]
Ok,  if the user deletes / update, you want to delete/update the feed.  I think we don't need sync up everything As feed is not a source of truth.

**Comment by cmanda** [comment]
Agreed on the source of truth - its about how we count activities under the leaderboard metrics.

**Comment by rathnapandi** [comment]
We are creating one record in feed irrespective of create/update activities. not sure what is missing here.
